src/kibana/components/vis/vis.js

```diff
@@ -102,6 +102,14 @@ define(function (require) {
       }
     };
 
+    Vis.prototype.hasSchemaAgg = function (schemaName, aggTypeName) {
+      var aggs = this.aggs.bySchemaName[schemaName] || [];
+      return aggs.some(function (agg) {
+        if (!agg.type || !agg.type.name) return false;
```

Comment:
LGTM. It is quite possible for `agg.type` to be undefined. `type.name` on the other hand should never be undefined, but doesn't hurt to check.